File: TimelineView/src/main/java/com/yogeshpaliyal/timelineview/Thumb.java
package com.yogeshpaliyal.timelineview;

import android.graphics.drawable.Drawable;

import androidx.annotation.Nullable;

public class Thumb {
    private float value;
    private float position;
    private Drawable drawable;

    public Thumb(){
        value = 0;
        position = 0;
    }

    public Thumb(@Nullable Drawable drawable) {
        value = 0;
        position = 0;
        // Clone the drawable so we can set the states individually
        this.drawable = drawable;
    }

    public Drawable getDrawable() {
        return drawable;
    }

    public void setDrawable(Drawable drawable) {
        this.drawable = drawable;
    }

    public float getPosition() {
        return position;
    }

    public void setPosition(float position) {
        this.position = position;
        // Update value based on new position
        this.value = position;
    }

    public float getValue() {
        return value;
    }

    public void setValue(float value) {
        this.value = value;
        this.position = value;
    }

}

File: TimelineView/src/main/java/com/yogeshpaliyal/timelineview/Thumb.kt
package com.yogeshpaliyal.timelineview

import android.graphics.drawable.Drawable

class Thumb {
    private var value: Float
    private var position: Float
    var drawable: Drawable? = null

    constructor() {
        value = 0f
        position = 0f
    }

    constructor(drawable: Drawable?) {
        value = 0f
        position = 0f
        // Clone the drawable so we can set the states individually
        this.drawable = drawable
    }

    fun getPosition(): Float {
        return position
    }

    fun setPosition(position: Float) {
        this.position = position
        // Update value based on new position
        value = position
    }

    fun getValue(): Float {
        return value
    }

    fun setValue(value: Float) {
        this.value = value
        position = value
    }
}